src/daft-core/src/datatypes/image_mode.rs

```diff
@@ -123,52 +123,6 @@ impl ImageMode {
     }
 }
 
-impl From<ImageMode> for image::ColorType {
-    fn from(image_mode: ImageMode) -> image::ColorType {
-        use image::ColorType;
-        use ImageMode::*;
-
-        match image_mode {
-            L => ColorType::L8,
-            LA => ColorType::La8,
-            RGB => ColorType::Rgb8,
-            RGBA => ColorType::Rgba8,
-            L16 => ColorType::L16,
-            LA16 => ColorType::La16,
-            RGB16 => ColorType::Rgb16,
-            RGBA16 => ColorType::Rgba16,
-            RGB32F => ColorType::Rgb32F,
-            RGBA32F => ColorType::Rgba32F,
-        }
-    }
-}
```

Comment:
Some of this stuff will conflict with https://github.com/Eventual-Inc/Daft/pull/2806  but I think we might want to prioritize merging this PR first since it seems we got rid of these entirely. Let's chat about it later with @samster25

Reply:
we can get #2806 in first, then i can rebase off of that